### PR TITLE
touchegg: 1.1.1 -> 2.0.8

### PR DESCRIPTION
**Pull request by Hvassaa**
###### Side note
This is my first PR to nixpkgs, so sorry if I am doing anything wrong! Also, if possible, I would like to maintain this package, but I do not know if there are any specific requirements to becoming a maintainer? Thanks.

###### Motivation for this change
Since touchegg started receiving updates again, the current touchegg in nixpkgs is quite outdated. 

###### Things done

There has been done some work upstream to make it possible to build on NixOS. See

* [Discussion at discourse](https://discourse.nixos.org/t/creating-a-simple-derivation-for-touchegg-but-the-binary-returns-an-error/11801)
* [PR on the touchegg project](https://github.com/JoseExposito/touchegg/pull/467)

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] (Found no tests) Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] (Found no packages that depended on touchegg with `grep -rnw '/path/to/nixpkgs/' -e 'touch.gg'`) Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Hvassaa** [comment]
The check from ofborg-eval `This PR does not cleanly list package outputs after merging.` gives the warning:
```Warning: `stdenv.lib` is deprecated and will be removed in the next release. Please use `lib` instead.```
I am not quite sure what the problem is. I tried fixing it in d54f18e

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:38]:
```suggestion
    platforms = platforms.linux;
```

stdenv.lib makes ofborg fail.

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:7]:
```suggestion
  pname = "touchegg";
  version = "2.0.8";

  src = fetchFromGitHub {
```

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:9]:
```suggestion
    repo = "${oname}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Hvassaa** [comment]
Thanks for reviewing!

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:8]:
```suggestion
    repo = pname;
```

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:29]:
```suggestion
    cairo
    gtk3-x11
    pcre
  ] ++ (with xorg; [
    libX11
    libXtst
    libXrandr
    libXi
    libXdmcp
    libpthreadstubs
    libxcb
  ]);
```

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:36]: gpl3Plus or gpl3Only?

**Comment by SuperSandro2000** [comment]
Also please squash the commits.

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (Hvassaa) [reply pkgs/tools/inputmethods/touchegg/default.nix]: Updated it to gpl3Plus

**Comment by Hvassaa** [comment]
> Also please squash the commits.

Done! :smile:

**Comment by CaptainSpof** [comment]
Touchégg v2.0.9 was released I got curious and found this PR.
It seems to be working great on my side (2.0.9), way better than my poor overlay attempt, thanks @Hvassaa

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:9]:
```suggestion
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:9]:
```suggestion
    rev = version;
```

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:10]:
```suggestion
    sha256 = "1adnqlb88yljwjybz0gsb948f7svyv01k2v5x2hhwgr0k85q8ic9";
```

**Comment by SuperSandro2000** [comment]
@Hvassaa the tests try to create a directory outside of the sandbox:

```
schemas/touchegg-2.0.8/glib-2.0/schemas/ install
[100%] Built target touchegg
Install the project...
-- Install configuration: "Release"
CMake Error at cmake_install.cmake:46 (file):
  file cannot create directory: /usr/share/touchegg.  Maybe need
  administrative privileges.
make: *** [Makefile:127: install] Error 1
builder for '/nix/store/j0khdssfdfq7bh7hn1hyz6s7q6mizhwr-touchegg-2.0.8.drv' failed with exit code 2
error: build of '/nix/store/j0khdssfdfq7bh7hn1hyz6s7q6mizhwr-touchegg-2.0.8.drv' failed
```

**Comment by arralka** [comment]
Once this is merged I'll have a go at packaging Touché.

**Comment by Hvassaa** [comment]
I think I will need help on this one, if anyone knows how to fix it

**Comment by astatide** [comment]
Looks like there's a bit of nasty business going on somewhere regarding the config.  I'm not an expert in cmake, but here's a grungy brute force "fix" added into the .nix file:

`  configurePhase = ''
    mkdir -p $out
    mkdir build
    cd build
    cmake -DCMAKE_INSTALL_PREFIX:PATH=$out -DCMAKE_BUILD_TYPE=Release ..

    # Not sure why the cmake paths aren't being set; there are, however, a few paths hard coded into files that need to be updated.
    sed -i "s|/lib|$out/lib|g" cmake_install.cmake
    sed -i "s|/etc|$out/etc|g" cmake_install.cmake
    sed -i "s|\''${CMAKE_INSTALL_PREFIX}|$out|g" cmake_install.cmake
    sed -i "s|/usr/share/|$out/share/|g" ../src/utils/paths.cpp
    sed -i "s|/usr/share/|$out/share/|g" ../src/utils/paths.h
    sed -i "s|/usr/share/|$out/share/|g" ../src/config/xml-config-loader.cpp
    sed -i "s|/usr/share/|$out/share/|g" ../src/config/xml-config-loader.h
  '';`

This at least gets it to install, be found in the path, and look for the configuration file in the correct place (touchegg blows up if it can't find it whatever directory was in src/utils.paths.cpp).  It does not get the systemd file to work.  

Hopefully this gives some insight as to why touchegg isn't installing and that there's a more elegant solution for at least the cmake portion that doesn't involve sed.  I just did a quick and dirty mod because I missed gestures in X

**Comment by astatide** [comment]
OH, it looks like the version that the .nix downloads is actually older than the PR you made, so it doesn't have any of those fixes in place.  Updating the file to version 2.0.10 (and updating the sha as appropriate) makes it work.  Here's a [gist with the updated file](https://gist.github.com/astatide/73649ae9c9e857486cbc454869aaeb3f).

**Comment by bobby285271** [comment]
Tracking touchegg 2.0.11 upgrade as part of Pantheon 6 upgrade in https://github.com/NixOS/nixpkgs/pull/130380#issuecomment-886060881.

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:37]: Missing maintainer.

File: pkgs/tools/inputmethods/touchegg/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/touchegg/default.nix:5]: 2.0.11 is the latest version.

**Comment by maydayv7** [comment]
Also, after this PR is merged, could the package be backported to 21.05, as GNOME 40 gestures in X11 requires the package

**Pull request closed without merging**
